include/picongpu/fields/FieldJ.kernel

```diff
@@ -131,12 +132,17 @@ namespace picongpu
             template<typename T_Particle, typename BoxJ, typename T_Acc>
             DINLINE void operator()(T_Acc const& acc, T_Particle& particle, BoxJ& jBox)
             {
-                const float_X weighting = particle[weighting_];
+                /* Use (potentially) damped weighting for charge calculation.
+                 * @see particles::boundary::DampWeightsInPml for details.
+                 */
+                const auto dampedWeighting = attribute::getDampedWeighting(particle);
```

Comment:
Maybe we should add an optimized function in a separate PR where we return the damped and nondamped at the same moment to avoid that we touch the global memory for the same value twice in case there is no damping factor and three times if there is a damping value.

Reply:
I think it should not really matter given the overall number of operations in current deposition. However, in principle indeed such a function can be written to load both a damped and undamped weight and do only minimal required number of global memory reads.